src/circuit/gadget/ecc/chip.rs

```diff
@@ -66,28 +72,44 @@ pub struct EccConfig {
 
     /// Incomplete addition
     pub q_add_incomplete: Selector,
+
     /// Complete addition
     pub q_add: Selector,
+
     /// Variable-base scalar multiplication (hi half)
     pub q_mul_hi: Selector,
     /// Variable-base scalar multiplication (lo half)
     pub q_mul_lo: Selector,
     /// Selector used in scalar decomposition for variable-base scalar mul
     pub q_mul_decompose_var: Selector,
+    /// Selector used in scalar decomposition for variable-base scalar mul
+    pub q_init_z: Selector,
     /// Variable-base scalar multiplication (final scalar)
-    pub q_mul_complete: Selector,
+    pub q_mul_z: Selector,
+    /// Variable-base scalar multiplication (overflow check)
+    pub q_mul_overflow: Selector,
+    // Fixed column to constrain s_{120..=125} of s = alpha + k_{254} * 2^127
+    pub q_three_bit: Selector,
+    // Fixed column to constrain s_{126} of s = alpha + k_{254} * 2^127
+    pub q_one_bit: Selector,
```

Comment:
It's simpler to treat the value as as being 130 bits rather than 127 bits, so that we don't need to special-case the top element. I've described this at https://hackmd.io/o9EzZBwxSWSi08kQ_fMIOw?both#Overflow-check-for-variable-base-scalar-multiplication .

```suggestion
```